templates/cluster-template.yaml

```diff
@@ -38,58 +38,64 @@ spec:
-            - hostPath: /etc/kubernetes/azure.json
-              mountPath: /etc/kubernetes/azure.json
-              name: cloud-config
-              readOnly: true
-      files:
-        - path: /etc/kubernetes/azure.json
-          owner: "root:root"
-          permissions: "0644"
-          content: |
-            {
-              "cloud": "AzurePublicCloud",
-              "tenantId": "${AZURE_TENANT_ID}",
-              "subscriptionId": "${AZURE_SUBSCRIPTION_ID}",
-              "aadClientId": "${AZURE_CLIENT_ID}",
-              "aadClientSecret": "${AZURE_CLIENT_SECRET}",
-              "resourceGroup": "${AZURE_RESOURCE_GROUP}",
-              "securityGroupName": "${CLUSTER_NAME}-controlplane-nsg",
-              "location": "${AZURE_LOCATION}",
-              "vmType": "standard",
-              "vnetName": "${CLUSTER_NAME}",
-              "vnetResourceGroup": "${CLUSTER_NAME}",
-              "subnetName": "${CLUSTER_NAME}-controlplane-subnet",
-              "routeTableName": "${CLUSTER_NAME}-node-routetable",
-              "userAssignedID": "${CLUSTER_NAME}",
-              "loadBalancerSku": "standard",
-              "maximumLoadBalancerRuleCount": 250,
-              "useManagedIdentityExtension": false,
-              "useInstanceMetadata": true
-            }
+    initConfiguration:
+      nodeRegistration:
+        name: '{{ ds.meta_data["local_hostname"] }}'
+        kubeletExtraArgs:
+          cloud-provider: azure
+          cloud-config: /etc/kubernetes/azure.json
+    joinConfiguration:
+      nodeRegistration:
+        name: '{{ ds.meta_data["local_hostname"] }}'
+        kubeletExtraArgs:
+          cloud-provider: azure
+          cloud-config: /etc/kubernetes/azure.json
+    clusterConfiguration:
+      apiServer:
+        timeoutForControlPlane: 20m
+        extraArgs:
+          cloud-provider: azure
+          cloud-config: /etc/kubernetes/azure.json
+        extraVolumes:
+          - hostPath: /etc/kubernetes/azure.json
+            mountPath: /etc/kubernetes/azure.json
+            name: cloud-config
+            readOnly: true
+      controllerManager:
+        extraArgs:
+          cloud-provider: azure
+          cloud-config: /etc/kubernetes/azure.json
+          allocate-node-cidrs: "false"
+        extraVolumes:
+          - hostPath: /etc/kubernetes/azure.json
+            mountPath: /etc/kubernetes/azure.json
+            name: cloud-config
+            readOnly: true
+    files:
+      - path: /etc/kubernetes/azure.json
+        owner: "root:root"
+        permissions: "0644"
+        content: |
+          {
+            "cloud": "AzurePublicCloud",
+            "tenantId": "${AZURE_TENANT_ID}",
+            "subscriptionId": "${AZURE_SUBSCRIPTION_ID}",
+            "aadClientId": "${AZURE_CLIENT_ID}",
+            "aadClientSecret": "${AZURE_CLIENT_SECRET}",
+            "resourceGroup": "${AZURE_RESOURCE_GROUP}",
+            "securityGroupName": "${CLUSTER_NAME}-controlplane-nsg",
```

Comment:
this needs to be `${CLUSTER_NAME}-node-nsg`

Reply:
this is the control plane kubeadm config

Reply:
fixed the routetable name

Reply:
kubernetes only makes modifications related to worker nodes (like creation of loadbalancer, opening up the nsg for specific ports etc). azure.json is meant to be used for controller manager as specified on line 70. So all the parameters here should be pointing to worker node parameters like `${CLUSTER_NAME}-node-subnet`, `${CLUSTER_NAME}-node-routetable` and `${CLUSTER_NAME}-node-nsg`. Since this already merged can you please correct in another PR. Let me know if you need further clarifications.